src/protocols/secure_channel/CASESession.cpp

```diff
@@ -335,57 +320,50 @@ CHIP_ERROR CASESession::DeriveSecureSession(SecureSession & session, SecureSessi
 
 CHIP_ERROR CASESession::SendSigmaR1()
 {
-    uint16_t data_len =
-        static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + sizeof(uint16_t) +
-                              mOpCredSet->GetCertCount() * kTrustedRootIdSize + kP256_PublicKey_Length + sizeof(uint64_t) * 4);
+    uint16_t data_len = static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + kMAX_Hash_Length +
+                                              kP256_PublicKey_Length + sizeof(uint64_t) * 4);
 
     System::PacketBufferTLVWriter tlvWriter;
     System::PacketBufferHandle msg_R1;
     TLV::TLVType outerContainerType                      = TLV::kTLVType_NotSpecified;
     uint8_t initiatorRandom[kSigmaParamRandomNumberSize] = { 0 };
+    uint8_t destinationIdentifier[kMAX_Hash_Length]      = { 0 };
+    // TODO: Parameterize fabricId instead of hardcoding it to 0
```

Comment:
This is important, because we have to make sure both sides share correct fabric ID, rather than assume it. This has to be validated during CASE when processing R2/R3

Reply:
Updated.